File: Python/백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
import sys

a,b = map(int,sys.stdin.readline().split())
aa = list(map(int,sys.stdin.readline().split()))
total = 0
add = []
for i in range(a):
    total+=aa[i]
    add.append(total)
for i in range(b):
    s,e = map(int,sys.stdin.readline().split())
    if s<2:
        sys.stdout.write(f"{add[e-1]}\n")
    else:
        sys.stdout.write(f"{add[e-1]-add[s-2]}\n")

File: C#/백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.cs
using System;
using System.Text;

namespace q{
    class w{
        static void Main(){
            StringBuilder sb = new StringBuilder();
            int[] q = Array.ConvertAll(Console.ReadLine().Split(),int.Parse);
            int a = q[0]; int b = q[1];
            int[] aa = Array.ConvertAll(Console.ReadLine().Split(),int.Parse);
            int[] sum = new int[a];
            int total = 0;
            for (int i=0;i<a;i++){
                total+=aa[i];
                sum[i] = total;
            }
            for (int i=0;i<b;i++){
                int[] bb = Array.ConvertAll(Console.ReadLine().Split(),int.Parse);
                int s = bb[0]; int e = bb[1];
                if (s<2){
                    sb.AppendLine(sum[e-1].ToString());
                }
                else{
                    sb.AppendLine((sum[e-1]-sum[s-2]).ToString());
                }
            }
            Console.WriteLine(sb.ToString());
        }
    }
}